ruby/lib/ci/queue/redis/worker.rb

```diff
@@ -13,9 +13,10 @@ class << self
       class Worker < Base
         attr_reader :total
 
-        def initialize(redis, config)
-          @reserved_test = nil
+        def initialize(redis, config, max_reserved_tests: 1)
```

Comment:
@ChrisBr I added a parameter we can use to ensure we are not reserving more tests than we allow in the in-memory queue. This is fairly easy and I think would catch the obvious issues? It would be a duplicate check since we already check this in the parallelization server but can't be too careful.